test/x509_test.c

```diff
@@ -276,12 +277,126 @@ static int test_x509_revoked_delete_last_extension(void)
 
     ret = 1;
 
-err:
+ err:
     X509_REVOKED_free(rev);
     X509_EXTENSION_free(ext);
     return ret;
 }
 
+static int test_drop_empty_cert_keyids(void)
```

Comment:
This tests the degenerate case mentioned by @bernd-edlinger of a self-signed cert with subjectKeyIdentifier=none and authorityKeyIdentifier=keyid, issuer.
Other more typical cases were already covered at CLI level by `test/recipes/25-test_req.t‎`
Doing the new test case there would have been much simpler, and IMO sufficient.

Reply:
Sorry, this is an initial minimal test case for the underlying `v2i` interfaces, more should be added.

Reply:
Sure, if you see the need to spend the effort to do more API-level tests, go for them.
I just wanted to make life easier - not only for you, but also for reviewers and maintainers.